v2/internal/testcommon/test_context.go

```diff
@@ -54,14 +54,28 @@ type PerTestContext struct {
 	Namespace           string
 }
 
-// If you modify this make sure to modify the cleanup-test-azure-resources target in the Makefile too
+// There are two prefixes here because each represents a resource kind with a distinct lifecycle.
+// If you modify these, make sure to modify the cleanup-azure-resources target in the Taskfile too
+
+// ResourcePrefix is for resources which are used in the record/replay test infrastructure. These tests are not expected to
+// be run live in parallel. In parallel runs should be done from the recordings. As such, for cleanup, we can delete any resources
```

Comment:
Is this going to cause problems with recording (against the ASO-CI sub)? Or do we just need to check that the live envtest action isn't running?

Reply:
This has always been the case, this is just documenting it.

Reply:
To answer the actual question, as long as the live envtest isn't running you should be fine - and we moved that to run only at like 4am PST or something so it should always be ok during working hours. We could collide with one another I suppose.

Reply:
Ah, I guess the specific thing I was wondering about was for new tests, which would never collide with existing live test runs since the group would be different, but the backstop delete job could remove them while the test was running. But you're right, if they're running in the middle of the night it's not really a worry.